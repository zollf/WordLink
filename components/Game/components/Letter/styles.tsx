import { StyleSheet } from "react-native";
import { widthPercentageToDP, heightPercentageToDP } from 'react-native-responsive-screen';

import AppStyle from "../../../../styles";

const styles = StyleSheet.create({
  box: {
    width: widthPercentageToDP('10%'),
    height: widthPercentageToDP('10%'),
    backgroundColor: AppStyle.lightGrey,
    margin: 5,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2},
    shadowOpacity: 0.1,
    borderRadius: 5,
  },
  text: {
    fontFamily: AppStyle.font,
    fontSize: heightPercentageToDP('3%'),
  },
});

export default styles;