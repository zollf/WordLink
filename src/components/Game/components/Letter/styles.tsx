import { StyleSheet } from 'react-native';
import { widthPercentageToDP, heightPercentageToDP } from 'react-native-responsive-screen';

import AppStyle from 'app/src/styles';

const styles = StyleSheet.create({
  box: {
    width: widthPercentageToDP('10%'),
    height: widthPercentageToDP('10%'),
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    margin: 3,
  },
  text: {
    fontFamily: AppStyle.fontBold,
    fontSize: heightPercentageToDP('3%'),
  },
  shadow: {
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2 },
    shadowOpacity: 0.05,
  },
});

export default styles;
