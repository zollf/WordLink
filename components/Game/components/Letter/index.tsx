import React from 'react';
import { Animated, TouchableOpacity, Text, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useAnimate } from 'app/hooks';

import { getGradient } from 'app/styles';
import styles from './styles';

interface Props {
  letter: string;
  color?: string;
  onPress?: () => void;
}

const Letter = ({ letter, color, onPress }: Props) => {
  const [spring] = useAnimate(1);

  const textColor = color ? 'white' : 'black';

  const handlePress = () => {
    Animated.sequence([
      Animated.timing(spring, {
        toValue: 0.8,
        duration: 1,
        useNativeDriver: true,
      }),
      Animated.spring(spring, {
        toValue: 1,
        friction: 3,
        useNativeDriver: true,
      }),
    ]).start();
    onPress && onPress();
  };

  return !onPress ? (
    <View data-test-id="static" style={styles.shadow}>
      <LinearGradient colors={getGradient(color)} style={styles.box} locations={[0.1, 0.8]} start={[-0.1, 0.6]}>
        <Text
          style={[
            styles.text,
            {
              color: textColor,
            },
          ]}
        >
          {letter}
        </Text>
      </LinearGradient>
    </View>
  ) : (
    <Animated.View
      style={{
        transform: [{ scale: spring }],
      }}
    >
      <TouchableOpacity data-test-id="letter-button" onPress={handlePress} style={styles.shadow}>
        <LinearGradient colors={getGradient(color)} style={styles.box} locations={[0.1, 0.8]} start={[-0.1, 0.6]}>
          <Text
            style={[
              styles.text,
              {
                color: textColor,
              },
            ]}
          >
            {letter}
          </Text>
        </LinearGradient>
      </TouchableOpacity>
    </Animated.View>
  );
};

export default Letter;
