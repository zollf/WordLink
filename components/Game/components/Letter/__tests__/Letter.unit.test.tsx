import React from 'react';
import useWithStores from 'app/test/useWithStores';

import Letter from '../';

import renderer from 'react-test-renderer';

const onPress = jest.fn();

const component = useWithStores(<Letter letter="L" color="orange" onPress={onPress} />);

describe('<Letter />', () => {
  let wrapper: renderer.ReactTestRenderer;
  beforeEach(() => {
    wrapper = renderer.create(component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click letter button calls function', () => {
    wrapper.root.findByProps({ 'data-test-id': 'letter-button' }).props.onPress();
    expect(onPress).toBeCalled();
  });

  it('static letter', () => {
    wrapper = renderer.create(useWithStores(<Letter letter="L" />));
    expect(wrapper.root.findAllByProps({ 'data-test-id': 'static' })).toBeDefined();
  });
});
