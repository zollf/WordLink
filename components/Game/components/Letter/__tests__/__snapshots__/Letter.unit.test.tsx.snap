// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Letter /> matches snapshot 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "scale": mockConstructor {},
        },
      ],
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 4,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4294956160,
            4294950464,
          ],
          "endPoint": undefined,
          "locations": Array [
            0.1,
            0.8,
          ],
          "startPoint": Array [
            -0.1,
            0.6,
          ],
        }
      }
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "display": "flex",
          "height": 75,
          "justifyContent": "center",
          "margin": 5,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Raleway_700Bold",
              "fontSize": 40,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        L
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
